### PR TITLE
PSR-7 Part 3: Post Bugfix & HTTPRequestOptions

### LINKED ISSUE #10475: [2021.09] PHP Warnings
Please report here any PHP Warning message obtained running the 2021.01 stable or 2021.06-dev versions.

Closes #10168

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp #10633 

- Bugfix `IHTTPClient::post()` (used wrong option for sending data in case it's XML instead of JSON ..)
- Introduce `HTTPRequestOptions` constants (because I always had to do deep-searches for finding all usages and references ...)
- Rename `DI::httpRequest()` into `DI::httpClient()`
- **Really** fixes https://github.com/friendica/friendica/issues/10475#issuecomment-905754227

Needs https://github.com/friendica/friendica-addons/pull/1161 as well

### CODE REVIEW

File: src/Network/HTTPRequestOptions.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Friendica\Network;
+
+use GuzzleHttp\RequestOptions;
+
+/**
+ * This class contains a list of possible HTTPClient request options.
+ */
+class HTTPRequestOptions
```
Comment (MrPetovan) [inline src/Network/HTTPRequestOptions.php]: According to the comment above, and to clear out confusion:
```suggestion
 * This class contains a list of possible HTTPClient request options.
 */
class HTTPClientOptions
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Network/HTTPRequestOptions.php (hunk 1 above)
Comment (nupplaphil) [reply src/Network/HTTPRequestOptions.php]: But in fact they're really request options (HTTPClient request options) ;-)

Comment (nupplaphil) [reply src/Network/HTTPRequestOptions.php]: OK, I changed it to `HTTPClientRequestOptions` :D

Comment (MrPetovan) [reply src/Network/HTTPRequestOptions.php]: Close, but not 1:1 so they are client options that we map to request options.

Comment (nupplaphil) [reply src/Network/HTTPRequestOptions.php]: hmm .. not fully convinced, but it's good for me :-)

**Review by MrPetovan — APPROVED** [review]

**Comment by nupplaphil** [comment]
ah, one moment, another "quick-update" --> rename `DI::httpRequest()` into `DI::httpClient()` ;-)

**Comment by nupplaphil** [comment]
the perfect place for such changes :D .. Renaming it inside another PRs would have made every reviewer mad ;-)

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**